TypeCobol/Compiler/Sql/Model/TableReference.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.CodeElements;
+
+namespace TypeCobol.Compiler.Sql.Model
+{
+    public abstract class TableReference
+    {
+        protected TableViewCorrelationName TableRef {get; set;}
```

Comment:
Instead of having a protected setter, you should use a public readonly property and set the value through the constructor. The base class will pass the value at construction time when calling the base constructor.

Reply:
In fact it does not matter as not every TableReference have a table ref. In this PR, I think we should go for:

```csharp
public class SingleTableOrViewReference : SqlObject
{
    public TableViewCorrelationName TableOrViewName { get; }
    public CorrelationClause { get; }
    //+ constructor and visitor
}
```

Reply:
Maybe we'll introduce a common ancestor class TableReference later